### DISCUSSION TITLE
Uploading files through Octokit throwing Bad Content Length Error

### DISCUSSION THREAD

**Discussion by divyam234** [octokit/.github#43] [Q&A]
```js
  const name=''
  const filePath=''
  const file = await Deno.stat(filePath);
  const body = (await Deno.open(filePath, { read: true })).readable;
  
  const res = await octokit.request('POST /repos/{owner}/{repo}/releases/{release_id}/assets{?name}', {
    owner: '',
    repo: '',
    release_id: '',
    name,
    data: body,
    headers: {
      'X-GitHub-Api-Version': '2022-11-28',
      'content-type': 'application/octet-stream',
      'content-length': file.size
    }
  })
```
This piece of code is throwing bad content length error.Although if I pass ArrayBuffer instead of Readable Stream its working

**Comment by gr2m** [comment]
Can you try not set `content-length` yourself? It will be set automatically.

**Reply by divyam234** [reply]
Same error if I omit content-length.I think problem lies when body is passed as readable stream. I tried same code in nodejs  its working there but not in deno .I think something related to fetch client of deno is causing problem. Let me post the same in deno group.

**Reply by gr2m** [reply]
If you find out what the problem was and if there is something we could do in https://github.com/octokit/request.js to normalize the behavior across JavaScript runtimes, please let us know

**Reply by divyam234** [reply]
Sure
